pkgs/applications/science/programming/fdr/default.nix

```diff
@@ -0,0 +1,87 @@
+{stdenv, fetchurl, qt57, ncurses, xorg, zlib, python27Packages}:
+let libs = [
+  stdenv.cc.cc
+  python27Packages.python
+  qt57.qtbase
+  qt57.qtx11extras
+  ncurses
+  xorg.libX11
+  xorg.libXft
+  zlib
+  ];
+in
+stdenv.mkDerivation {
+  name = "fdr";
+  src = fetchurl {
+    url = https://www.cs.ox.ac.uk/projects/fdr/downloads/fdr-3754-linux-x86_64.tar.gz;
+    md5 = "c09a05a5d9b54fc65ed12de8b7cdfc11";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+  installPhase = ''
+    mkdir -p "$out"
+
+    # shipped Qt is buggy
+    rm lib/libQt*
+    rm -r lib/qt_plugins
+
+    cp -r * "$out"
+    # Hack around lack of libtinfo in NixOS
+    ln -s ${ncurses.out}/lib/libncursesw.so.6 $out/lib/libtinfo.so.5
+    ln -s ${qt57.qtbase.out}/lib/qt5/plugins $out/lib/qt_plugins
+    ln -s ${zlib.out}/lib/libz.so.1 $out/lib/libz.so.1
+
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" "$out/bin/fdr4"
```

Comment:
Instead of copying around the `stdenv.lib.makeLibraryPath`, you may do it at the top (I'll leave a comment there) and just reference libs. Especially true if you don't need $out in it.